src/coreclr/jit/fgbasic.cpp

```diff
@@ -684,24 +683,48 @@ void Compiler::fgReplaceJumpTarget(BasicBlock* block, BasicBlock* oldTarget, Bas
                 {
                     assert(block->KindIs(BBJ_COND));
                     block->SetTrueEdge(newEdge);
-
-                    if (oldEdge->hasLikelihood())
-                    {
-                        newEdge->setLikelihood(oldEdge->getLikelihood());
-                    }
                 }
             }
             else
             {
                 assert(block->FalseTargetIs(oldTarget));
                 FlowEdge* const oldEdge = block->GetFalseEdge();
 
+                if (block->TrueEdgeIs(oldEdge))
```

Comment:
I'm a bit confused by this check here. If a block's true and false edges are the same, then shouldn't the condition `if (block->TrueTargetIs(oldTarget))` above be true, making this code unreachable?